PowerArubaCP/Public/VM.ps1

```diff
@@ -0,0 +1,751 @@
+
+        Initial configuration with minimal settings of ClearPasss (Name, IP Address MGMT, DNS, Password...)
+
+        .EXAMPLE
+        $cppmsetupParams = @{
+            name_vm                 = "CPPM"
+            version                 = "6.9"
+            hostname                = "CPPM"
+            mgmt_ip                 = "192.0.2.225"
+            mgmt_netmask            = "24"
+            mgmt_gateway            = "192.0.2.254"
+            data_ip                 = "198.0.51.225"
+            data_netmask            = "24"
+            data_gateway            = "198.0.51.254"
+            dns_primary             = "192.0.2.1"
+            dns_secondary           = "198.0.51.200"
+            new_password            = "MyPassword"
+            ntp_primary             = "192.0.2.1"
+            ntp_secondary           = "198.0.51.200"
+            timezone_continent      = "8"
+            timezone_country        = "14"
+        }
+
+        PS>Set-ArubaCPVmSetup @cppmsetupParams
+
+        Initial configuration of ClearPasss (Name, IP Address MGMT/Data, DNS, Password, NTP, Timezone...)
+        for Timezone, on this example, it is for Europe/France(Paris)
+    #>
+
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingPlainTextForPassword", "")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    Param(
+        [string]$name_vm,
+        [Parameter (Mandatory = $true)]
+        [ValidateSet("6.8", '6.9')]
+        [string]$version,
+        [Parameter (Mandatory = $true)]
+        [string]$hostname,
+        [Parameter (Mandatory = $true)]
+        [ipaddress]$mgmt_ip,
+        [Parameter (Mandatory = $true)]
+        [ValidateRange(0, 32)]
+        [int]$mgmt_netmask,
+        [Parameter (Mandatory = $true)]
+        [ipaddress]$mgmt_gateway,
+        [Parameter (Mandatory = $false)]
+        [ipaddress]$data_ip,
+        [ValidateRange(0, 32)]
+        [int]$data_netmask,
+        [Parameter (Mandatory = $true)]
+        [ipaddress]$data_gateway,
+        [Parameter (Mandatory = $true)]
+        [ipaddress]$dns_primary,
+        [Parameter (Mandatory = $false)]
+        [ipaddress]$dns_secondary,
+        [Parameter (Mandatory = $true)]
+        [string]$new_password,
+        [Parameter (Mandatory = $false)]
+        [ValidateRange(1, 10)]
+        [int]$timezone_continent,
+        [Parameter (Mandatory = $false)]
+        [ValidateRange(1, 55)]
+        [int]$timezone_country,
+        [Parameter (Mandatory = $false)]
+        [ipaddress]$ntp_primary,
+        [Parameter (Mandatory = $false)]
+        [ipaddress]$ntp_secondary,
+        [Parameter (Mandatory = $false)]
+        [switch]$fips
+    )
+
+    Begin {
+    }
+
+    Process {
+        if ($timezone_continent -xor $timezone_country) {
+            Throw "You need to specific Timezone Continent and Country on the sametime"
```

Comment:
same time

Reply:
You need timezone continent and country (too complicated to use ParameterSetName on this case)